### PR TITLE
use the max_packet_size transport parameter

### LINKED ISSUE #1233: use the max_packet_size transport parameter
We're currently sending the max_packet_size transport parameter, but we're not reading or acting on it.

### PR DESCRIPTION
**Pull request by marten-seemann**
Fixes #1233.

### CODE REVIEW

File: internal/handshake/transport_parameters.go [hunk 1]
```diff
@@ -137,6 +139,15 @@ func readTransportParamters(paramsList []transportParameter) (*TransportParamete
 				return nil, fmt.Errorf("wrong length for omit_connection_id: %d (expected empty)", len(p.Value))
 			}
 			params.OmitConnectionID = true
+		case maxPacketSizeParameterID:
+			if len(p.Value) != 2 {
+				return nil, fmt.Errorf("wrong length for max_packet_size: %d (expected 2)", len(p.Value))
+			}
+			maxPacketSize := protocol.ByteCount(binary.BigEndian.Uint16(p.Value))
+			if maxPacketSize < 1200 {
+				return nil, fmt.Errorf("invalid value for max_packet_size: %d (minimum 1200)", maxPacketSize)
```
Comment (lucas-clemente) [inline internal/handshake/transport_parameters.go]: Should this be a qerr?

Comment (marten-seemann) [reply internal/handshake/transport_parameters.go]: It doesn't matter. This will end up as a TLS Internal Error alert anyway.

**Comment by marten-seemann** [comment]
We allocate buffers with the `MaxReceivePacketSize`.
The maximum size we use for packing packets is `MaxPacketSizeIPv4`, which is smaller than the `MaxReceivePacketSize`. The parameter sent in the TLS transport parameters can only lead to a reduction in the packet size for packets that we send, see https://github.com/lucas-clemente/quic-go/pull/1237/files#diff-4edc6b25b26b03f5ea2847eb0d3487e8R524.

**Comment by lucas-clemente** [comment]
ic, thx for clarifying.

**Pull request merged**